GitUI/CommandsDialogs/SettingsDialog/MergeToolsHelper.cs

```diff
@@ -41,44 +41,49 @@ public static string FindFileInFolders(string fileName, params string[] location
 
                 string programFilesPath = Environment.GetEnvironmentVariable("ProgramFiles");
 
-                string path;
-
-                if (!string.IsNullOrEmpty(programFilesPath))
+                if (CheckFileExists(programFilesPath, location, fileName, out string fullName))
                 {
-                    path = Path.Combine(programFilesPath, location);
-                    if (Directory.Exists(path))
-                    {
-                        string fullName = Path.Combine(path, fileName);
-                        if (File.Exists(fullName))
-                        {
-                            return fullName;
-                        }
-                    }
+                    return fullName;
                 }
 
                 if (IntPtr.Size == 8
                     || (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable("PROCESSOR_ARCHITEW6432"))))
                 {
                     programFilesPath = Environment.GetEnvironmentVariable("ProgramFiles(x86)");
 
-                    if (!string.IsNullOrEmpty(programFilesPath))
+                    if (CheckFileExists(programFilesPath, location, fileName, out fullName))
                     {
-                        path = Path.Combine(programFilesPath, location);
-                        if (Directory.Exists(path))
-                        {
-                            string fullName = Path.Combine(path, fileName);
-                            if (File.Exists(fullName))
-                            {
-                                return fullName;
-                            }
-                        }
+                        return fullName;
                     }
                 }
+
+                string localAppDataPath = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
+
+                if (CheckFileExists(localAppDataPath, location, fileName, out fullName))
+                {
+                    return fullName;
+                }
             }
 
             return string.Empty;
         }
 
+        private static bool CheckFileExists(string path, string location, string fileName, out string fullName)
```

Comment:
Please inline the method into the caller and get rid of `out string fullName` parameter. Then the method can be simplified down to the following:

```
bool CheckFileExists(string path)
{
    fullName = Path.Combine(path, location, fileName);
    return File.Exists(fullName);
}
```

Reply:
Agree that it would be nicer as a local function.

Be careful though about reassigning `fullName` as it can create problems, as discussed ⬆️ .

Reply:
Done.